src/assets.ts

```diff
@@ -0,0 +1,148 @@
+import { relative, resolve } from 'path'
+import fs from 'fs'
+import crypto from 'crypto'
+import { GenerateSWConfig, InjectManifestConfig, ManifestEntry } from 'workbox-build'
+import { ResolvedConfig } from 'vite'
+import { ResolvedVitePWAOptions } from './types'
+import { FILE_MANIFEST } from './constants'
+
+function addManifestEntry(
+  publicDir: string,
+  path: string,
+  includeUrl: string[],
+  additionalManifestEntries: ManifestEntry[],
+) {
+  let usePath
+  // chek only for relative paths
+  if (path.startsWith('/'))
+    usePath = resolve(publicDir, path.substring(1))
+  else
+    usePath = resolve(publicDir, path)
+
+  const url = relative(publicDir, usePath)
+  if (!includeUrl.includes(url) && fs.existsSync(usePath)) {
+    const cHash = crypto.createHash('MD5')
+    cHash.update(fs.readFileSync(usePath))
+    additionalManifestEntries.push({
+      url,
+      revision: `${cHash.digest('hex')}`,
+    })
+    includeUrl.push(url)
+  }
+}
+
+function addWebManifestEntry(
+  options: ResolvedVitePWAOptions,
+  includeUrl: string[],
+  additionalManifestEntries: ManifestEntry[],
+) {
+  if (!includeUrl.includes(FILE_MANIFEST)) {
+    const cHash = crypto.createHash('MD5')
+    cHash.update(generateWebManifestFile(options))
+    additionalManifestEntries.push({
+      url: FILE_MANIFEST,
+      revision: `${cHash.digest('hex')}`,
+    })
+    includeUrl.push(FILE_MANIFEST)
+  }
+}
+
+function resolveAdditionalManifestEntries(
+  useInjectManifest: boolean,
+  includeUrl: string[],
+  injectManifest: Partial<InjectManifestConfig>,
+  workbox: Partial<GenerateSWConfig>,
+): ManifestEntry[] {
+  let additionalManifestEntries: ManifestEntry[]
+
+  if (useInjectManifest)
+    additionalManifestEntries = (injectManifest.additionalManifestEntries = injectManifest.additionalManifestEntries || [])
+
+  else
+    additionalManifestEntries = (workbox.additionalManifestEntries = workbox.additionalManifestEntries || [])
+
+  if (additionalManifestEntries.length > 0) {
+    additionalManifestEntries
+      .filter(me => !includeUrl.includes(me.url))
+      .forEach(me => includeUrl.push(me.url))
+  }
+  return additionalManifestEntries
+}
+
+export function configureStaticAssets(
+  resolvedVitePWAOptions: ResolvedVitePWAOptions,
+  viteConfig: ResolvedConfig,
+) {
+  const {
+    manifest,
+    strategies,
+    injectManifest,
+    workbox,
+    include,
+    includeManifestIcons,
+  } = resolvedVitePWAOptions
+
+  // static assets handling
+  // we need to check inject manisfest strategy
+  // additionalManifestEntries will go to workbox entry
+  // or to injectManifest entry
+  const { publicDir } = viteConfig
+  const useInjectManifest = strategies === 'injectManifest'
+  // include static assets
+  const includeUrl: string[] = []
```

Comment:
modify `include` type adding `false`, defaults to undefined so we need to check if `undefined` (I'll review defaults to also include some apple icons, see picture below):

`['favicon.ico', 'favicon-<size>.png', 'favicon.svg', 'safari-pinned-tab.svg'>]`

where `<sizes>` will be: `16x16`,`32x32`, `48x48`, `72x72`, `96x96`

then merge with provided on `include` option.

We also need to add this to `README.md` file.

![imagen](https://user-images.githubusercontent.com/6311119/118411808-c1efcd00-b696-11eb-8159-d260cd35a618.png)

Reply:
If we include these defaults, then update `vue-router` and `inject-manifest` examples removingth `favicon.svg` from `include` option.

Reply:
maybe it will be better to do something similar to Vite `assets` plugin (see [assetsInclude configuration option](https://vitejs.dev/config/#assetsinclude)): `include: string | Regex | (string | Regex)[] | undefined` and force the user to include all static assets.

Reply:
Could simply accepts an array of globs, and pass it to `fast-glob` to get the file list. Also for auto included icons, we could run a unique so ppl don't need to care if they should exclude them or not.

Reply:
But, how to resolve the url path added to the `__SW_MAINIFEST`? For manifest icons we can assume to be under Vite's `publicDir` option directory, but for `includeAssets`?

Reply:
yeah, both of them should be under the publicDir. Or do you see any other cases?

Reply:
ok, so we only need to check if under Vite's  `publucDir` once globs resolved?

Reply:
OK, we also need to change `configResolved` to be `async` on `index.ts`.

Reply:
If it's hard, we can use `fg.sync` for sync. But would prefer async for performance if possible.

Reply:
no problem, just adding `async` to `configResolved`and `resolveOptions` and done

Reply:
![imagen](https://user-images.githubusercontent.com/6311119/118798484-fc778680-b89d-11eb-900d-43afb8e8d4de.png)

Reply:
can I use `Promise.allSettled` to do md5 computation parallel or just sequential with `Promise.all`?